src/main/drivers/sbus_output.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * This file is part of Rotorflight.
+ *
+ * Rotorflight is free software. You can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Rotorflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "sbus_output.h"
+
+#include <math.h>
+#include <stdbool.h>
+#include <string.h>
+
+#include "build/build_config.h"
+#include "common/maths.h"
+#include "common/time.h"
+#include "flight/mixer.h"
+#include "io/serial.h"
+#include "pg/sbus_output.h"
+#include "platform.h"
+#include "rx/rx.h"
+
+STATIC_UNIT_TESTED serialPort_t *sbusOutPort = NULL;
+
+STATIC_UNIT_TESTED void sbusOutPrepareSbusFrame(sbusOutFrame_t *frame,
+                                                uint16_t *channels)
+{
+    frame->syncByte = 0x0F;
+
+    // There's no way to make a bit field array, so we have to go tedious.
+    frame->chan0 = channels[0];
+    frame->chan1 = channels[1];
+    frame->chan2 = channels[2];
+    frame->chan3 = channels[3];
+    frame->chan4 = channels[4];
+    frame->chan5 = channels[5];
+    frame->chan6 = channels[6];
+    frame->chan7 = channels[7];
+    frame->chan8 = channels[8];
+    frame->chan9 = channels[9];
+    frame->chan10 = channels[10];
+    frame->chan11 = channels[11];
+    frame->chan12 = channels[12];
+    frame->chan13 = channels[13];
+    frame->chan14 = channels[14];
+    frame->chan15 = channels[15];
+
+    frame->flags = channels[16] ? 0x01 : 0x00;
+    frame->flags += channels[17] ? 0x02 : 0x00;
```

Comment:
When dealing with bitmasks, don't use +, but |
```
frame->flags = channels[16] ? BIT(0) : 0;
frame->flags |= channels[17] ? BIT(1) : 0;
```

Reply:
Fixed

Reply:
Please use the BIT() marcro too!

Reply:
Missed it. [5cfeaea](https://github.com/rotorflight/rotorflight-firmware/pull/190/commits/5cfeaeaaf89d81549a126bd9bbd13597f22ca67a)